### PR TITLE
More centralized mutations

### PR DESCRIPTION
**Pull request by manzoorwanijk**
Completes #1983 

## Checklist

* [ ] I have read the documentation relating to systems affected by this pull request, see https://github.com/eventespresso/event-espresso-core/tree/master/docs
* [ ] User input is adequately validated and sanitized
* [ ] all publicly displayed strings are internationalized (usually using `esc_html__()`, see https://codex.wordpress.org/I18n_for_WordPress_Developers)
* [ ] My code is tested.
* [ ] My code follows the Event Espresso code style.
* [ ] My code has proper inline documentation.
* [ ] My code accounts for when the site is in Maintenance Mode (MM2 especially disallows usage of models)

### CODE REVIEW

File: assets/prototype/domain/eventEditor/containers/mutations/tickets/updatePriceCache.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+				where: {
+					ticketIn,
+				},
+			},
+		});
+	} catch (error) {
+		data = {};
+	}
+
+	const newTicketIn = remove ? ticketIn.filter((id) => id !== ticketId) : [...ticketIn, ticketId];
```
[the remark this answers is not in the document]
Comment (manzoorwanijk) [reply assets/prototype/domain/eventEditor/containers/mutations/tickets/updatePriceCache.js]: Agreed

File: assets/prototype/domain/eventEditor/containers/mutations/prices/useOnCreatePrice.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+			});
+			updateRelations({
+				entity: 'prices',
+				entityId: priceId,
+				relation: 'priceTypes',
+				relationIds: [priceTypeId],
+			});
+		}
+		// Update price cache.
+		updatePriceCache({ proxy, prices, price });
```
Comment (nicolad) [inline assets/prototype/domain/eventEditor/containers/mutations/prices/useOnCreatePrice.js]: Clean :+1:

File: assets/prototype/domain/eventEditor/containers/mutations/prices/useOnUpdatePrice.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import useRelations from '../../../../../infrastructure/services/relations/useRelations';
+
+const useOnUpdatePrice = () => {
+	const { updateRelations } = useRelations();
+
+	const onUpdatePrice = ({ price, priceTypeId }) => {
+		const { id: priceId } = price;
+		// if it's not the optimistic response
+		// and priceType has been updated.
+		if (priceId && priceTypeId) {
```
Comment (nicolad) [inline assets/prototype/domain/eventEditor/containers/mutations/prices/useOnUpdatePrice.js]: It's not entirely clear how this conditional is checking `if it's not the optimistic response`,
maybe create a variable with those conditions?

File: assets/prototype/domain/eventEditor/containers/mutations/prices/usePriceMutator.js [hunk 4]
```diff
@@ -0,0 +1,61 @@
+import usePriceQueryOptions from './usePriceQueryOptions';
+import useOnCreatePrice from './useOnCreatePrice';
+import useOnUpdatePrice from './useOnUpdatePrice';
+import useOnDeletePrice from './useOnDeletePrice';
+
+/**
+ *
+ */
```
Comment (nicolad) [inline assets/prototype/domain/eventEditor/containers/mutations/prices/usePriceMutator.js]: // Silence is golden.

File: assets/prototype/domain/eventEditor/containers/mutations/prices/useOnUpdatePrice.js (hunk 3 above)
Comment (manzoorwanijk) [reply assets/prototype/domain/eventEditor/containers/mutations/prices/useOnUpdatePrice.js]: An optimistic response won't have the `id` field if we create one.

File: assets/prototype/domain/eventEditor/containers/mutations/prices/usePriceMutator.js [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+		return {
+			input: mutationInput,
+		};
+	};
+
+	const mutator = (mutationType, input) => {
+		// `ticketId` must be passed for price creation
+		// so as to properly set the relations
+		const { ticketId, ...restInput } = input;
+		const { priceType: priceTypeId } = input;
```
Comment (nicolad) [inline assets/prototype/domain/eventEditor/containers/mutations/prices/usePriceMutator.js]: What do you think about this:
```suggestion
		const { ticketId, priceType: priceTypeId, ...restInput } = input;
```
?

Comment (manzoorwanijk) [reply assets/prototype/domain/eventEditor/containers/mutations/prices/usePriceMutator.js]: `priceType` is needed in `restInput`

Comment (nicolad) [reply assets/prototype/domain/eventEditor/containers/mutations/prices/usePriceMutator.js]: Or maybe use `omit` from ramda:
https://ramdajs.com/docs/#omit
in `createVariables`

Comment (nicolad) [reply assets/prototype/domain/eventEditor/containers/mutations/prices/usePriceMutator.js]: I understood you didn't want to pass entire `input` object.

Comment (manzoorwanijk) [reply assets/prototype/domain/eventEditor/containers/mutations/prices/usePriceMutator.js]: `ticketId` is still to be removed as it's not a part of the mutation schema 😄

File: assets/prototype/domain/eventEditor/containers/mutations/prices/useOnUpdatePrice.js (hunk 3 above)
Comment (nicolad) [reply assets/prototype/domain/eventEditor/containers/mutations/prices/useOnUpdatePrice.js]: Oh yeah, forgot about that :man_facepalming:

File: assets/prototype/domain/eventEditor/containers/mutations/tickets/updatePriceCache.js [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+
+	let newData;
+	if (!remove && prices.length) {
+		const pathToNodes = ['prices', 'nodes'];
```
Comment (nicolad) [inline assets/prototype/domain/eventEditor/containers/mutations/tickets/updatePriceCache.js]: Little nice and clean details :+1:

**Pull request merged**